app/markets/ActivityTable.tsx

```diff
@@ -33,33 +42,49 @@ const TAG_COLOR_SCHEMES: { 0: TagColorSchemeProp; 1: TagColorSchemeProp } = {
 
 const ITEMS_PER_PAGE = 10;
 
-export const ActivityTable = ({ id }: { id: string }) => {
-  const [page, setPage] = useState(1);
+type MergedTradeTransaction = Omit<FpmmTrade, '__typename'> &
+  Partial<Omit<FpmmTransaction, '__typename'>> & {
+    __typename?: 'FpmmTrade' | 'FpmmTransaction';
+  };
 
-  const { data: trades, isLoading } = useQuery({
-    queryKey: ['getMarketTrades', id, page - 1],
-    queryFn: async () =>
-      getMarketTrades({
-        first: ITEMS_PER_PAGE,
-        skip: (page - 1) * ITEMS_PER_PAGE,
-        fpmm: id,
-        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
-        orderDirection: OrderDirection.Desc,
-      }),
+function mergeMarketTradesAndTransactionsArrays(
+  trades: FpmmTrade[],
+  transactions: FpmmTransaction[]
+): MergedTradeTransaction[] {
+  const tradesAndTransactions = new Map<string, MergedTradeTransaction>();
+
+  trades.forEach(trade => {
+    tradesAndTransactions.set(trade.id, { ...trade, __typename: 'FpmmTrade' });
   });
 
-  const { data: tradesNextPage } = useQuery({
-    queryKey: ['getMarketTrades', id, page],
-    queryFn: async () =>
-      getMarketTrades({
-        first: ITEMS_PER_PAGE,
-        skip: page * ITEMS_PER_PAGE,
-        fpmm: id,
-        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
-        orderDirection: OrderDirection.Desc,
-      }),
+  transactions.forEach(transaction => {
+    if (tradesAndTransactions.has(transaction.id)) {
+      const trade = tradesAndTransactions.get(transaction.id)!;
```

Comment:
```suggestion
      const trade = tradesAndTransactions.get(transaction.id);
```

is this a typo?

Reply:
No. It's used to tell the TypeScript compiler that we're certain there's a value won't be null or undefined, even if TypeScript's type inference suggests it might be.